crates/llm-ls/src/adaptors.rs

```diff
@@ -0,0 +1,142 @@
+use super::{
+    internal_error, APIError, APIResponse, CompletionParams, Generation, Ide, NAME, VERSION,
+};
+use reqwest::header::{HeaderMap, HeaderValue, AUTHORIZATION, USER_AGENT};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use tower_lsp::jsonrpc;
+
+struct AdaptHuggingFaceRequest;
+impl AdaptHuggingFaceRequest {
+    fn adapt_body(&self, prompt: String, params: CompletionParams) -> Value {
+        return serde_json::json!({
+            "inputs": prompt,
+            "parameters": params.request_params,
+        });
+    }
+    fn adapt_headers(
+        &self,
+        api_token: Option<&String>,
+        ide: Ide,
+    ) -> Result<HeaderMap, jsonrpc::Error> {
+        let mut headers = HeaderMap::new();
+        let user_agent = format!("{NAME}/{VERSION}; rust/unknown; ide/{ide:?}");
+        headers.insert(
+            USER_AGENT,
+            HeaderValue::from_str(&user_agent).map_err(internal_error)?,
+        );
+
+        if let Some(api_token) = api_token {
+            headers.insert(
+                AUTHORIZATION,
+                HeaderValue::from_str(&format!("Bearer {api_token}")).map_err(internal_error)?,
+            );
+        }
+
+        Ok(headers)
+    }
+}
+
+struct AdaptHuggingFaceResponse;
```

Comment:
I would've done two adapters, one for TGI and one for the API inference (the `APIResponse::Generation` is TGI and `APIResponse::Generations` Api inference):
- `TgiResponse`
- `ApiInferenceResponse`